src/Components/Authentication/FormSwitcher.tsx

```diff
@@ -132,6 +132,19 @@ export class FormSwitcher extends React.Component<FormSwitcherProps, State> {
     )
   }
 
+  getEmailValue = (): string => {
+    const { values } = this.props
+    const isClient = typeof window !== "undefined"
+    let email
+
+    if (isClient) {
+      const searchQuery = window.location.search.slice(1)
+      email = qs.parse(searchQuery).email as string
```

Comment:
Presumably this does proper URL-decoding?

Reply:
Confirmed that this works with an encoded email string like `user+with-extra@example.com` / `user%2Bwith-extra%40example.com`

![Screenshot from 2020-03-20 14-32-01](https://user-images.githubusercontent.com/123595/77195182-9cc5f580-6ab7-11ea-981f-6f47f3b3334a.png)